From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Frajo Haider <f_haider@gmx.at>
Date: Thu, 22 Sep 2022 20:43:40 +0300
Subject: [PATCH] Add support for KeyboardDisplay type agents.

---
 src/agent.h     | 1 +
 src/manager.cpp | 3 +++
 2 files changed, 4 insertions(+)

diff --git a/src/agent.h b/src/agent.h
index 853f8ff..2b534f5 100644
--- a/src/agent.h
+++ b/src/agent.h
@@ -48,6 +48,7 @@ public:
         DisplayYesNo = 1,
         KeyboardOnly = 2,
         NoInputNoOutput = 3,
+        KeyboardDisplay = 4,
     };
     Q_ENUM(Capability)
 
diff --git a/src/manager.cpp b/src/manager.cpp
index 897555b..799e3dd 100644
--- a/src/manager.cpp
+++ b/src/manager.cpp
@@ -157,6 +157,9 @@ PendingCall *Manager::registerAgent(Agent *agent)
     case Agent::NoInputNoOutput:
         capability = QStringLiteral("NoInputNoOutput");
         break;
+    case Agent::KeyboardDisplay:
+        capability = QStringLiteral("KeyboardDisplay");
+        break;
     default:
         capability = QStringLiteral("DisplayYesNo");
         break;
